applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt_scan_for_conn.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bt_mgmt_scan_for_conn_internal.h"
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+
+#include "bt_mgmt.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(bt_mgmt);
+
+#define CONNECTION_PARAMETERS                                                                      \
+	BT_LE_CONN_PARAM(CONFIG_BLE_ACL_CONN_INTERVAL, CONFIG_BLE_ACL_CONN_INTERVAL,               \
+			 CONFIG_BLE_ACL_SLAVE_LATENCY, CONFIG_BLE_ACL_SUP_TIMEOUT)
+
+static uint8_t bonded_num;
+struct bt_le_scan_cb scan_callback;
+static bool cb_registered;
+static char const *srch_name;
+
+static void bond_check(const struct bt_bond_info *info, void *user_data)
+{
+	char addr_buf[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(&info->addr, addr_buf, BT_ADDR_LE_STR_LEN);
+
+	LOG_DBG("Stored bonding found: %s", addr_buf);
+	bonded_num++;
+}
```

Comment:
The ```if (bonded_num)``` is not needed, I agree, but the ```if (bonded_num < CONFIG_BT_MAX_PAIRED) {``` is to avoid connecting to an unpaired device if we have already filled up the slots. Do you mean it is not necessary?

Reply:
Discussed and did remove ``if (bonded_num)`` but kept ``if (bonded_num >= CONFIG_BT_MAX_PAIRED)``